libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/Actions/Microsoft.QnAMakerAction.schema

```diff
@@ -0,0 +1,56 @@
+{
+    "$schema": "https://raw.githubusercontent.com/microsoft/botbuilder-dotnet/4.Future/schemas/component.schema",
+    "$role": "unionType(Microsoft.IDialog)",
+    "title": "QnAMaker Action",
+    "description": "This represents an action which is driven by a call to QnAMaker.ai knowledge base",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+        "kb-ID": {
+            "type": "string",
+            "title": "Knowledgebase Id",
+            "description": "Kb ID of your QnA Maker KB.",
+            "default": "settings.qna.kbid"
+        },
+        "endpointKey": {
+            "type": "string",
+            "title": "Endpoint Key",
+            "description": "Endpoint key for the QnA Maker KB.",
+            "default": "settings.qna.endpointkey"
+        },
+        "hostname": {
+            "type": "string",
+            "title": "Host",
+            "description": "Hostname for your QnA Maker service.",
+            "default": "settings.qna.hostname",
+            "examples": [
+                "https://yourserver.azurewebsites.net/qnamaker"
+            ]
+        },
+        "threshold": {
+            "type": "float",
+            "title": "Threshold",
+            "description": "Threshold score to filter results.",
+            "default": 0.3
+        },
+        "strictFilterProperty": {
```

Comment:
there is no type on this field, I think it should be a string

Reply:
for the property with a "$role", it don't need a buit-in type.

---
In reply to: [321931800](https://github.com/microsoft/botbuilder-dotnet/pull/2469#discussion_r321931800) [](ancestors = 321931800)